chore: internal docs for beta

alvarosabu opened the pull request:
This PR contains the temporary documentation for the Storyblok CLI v4 beta. Eventually, this will be moved to the official documentation

alvarosabu committed:
docs: v4 beta internal docs

- Updated the main README to indicate that official documentation for package v4 is not yet available, directing users to the internal documentation for the v4 beta.
- Created a new README file in the src directory to provide detailed installation instructions and command references for the Storyblok CLI package.
- Added a comprehensive guide for the `login` command, including usage examples and options.

alvarosabu committed:
docs(logout): add README for logout command

- Created a new README file for the `logout` command, detailing its usage, functionality, and examples.
- Included information on credential removal, session clearing, and warnings for already logged-out users.
- Aimed to enhance user understanding and provide clear instructions for the logout process.

alvarosabu committed:
docs(languages): add README for languages command

- Created a new README file for the `languages` command, detailing its usage, options, and examples for pulling language configurations from a Storyblok space.
- Included information on file structure and notes for user requirements, enhancing clarity and usability for developers interacting with the command.

alvarosabu committed:
docs(user): add README for user command

- Created a new README file for the `user` command, detailing its usage, functionality, and examples for viewing user information in Storyblok.
- Updated the main README to include a reference to the new `user` command, enhancing clarity and usability for developers interacting with the CLI.
- Aimed to provide comprehensive guidance for users to understand and utilize the command effectively.

alvarosabu committed:
docs(README): update command reference and add global options section

- Updated the Command Reference section in the README to improve clarity and organization.
- Added a new Global Options section detailing options available for all commands, including descriptions and default values.
- Included a tip for users on providing verbose output when reporting bugs or support tickets to enhance troubleshooting.

alvarosabu committed:
docs: clean up README files by removing trailing whitespace

- Removed unnecessary trailing whitespace from the README files for the `languages`, `login`, and `user` commands to improve formatting and maintain consistency across documentation.
- Ensured that all files end with a newline for better compatibility with various text editors.

alvarosabu committed:
docs(components): add README files for components commands and update options

- Created detailed README files for the `components pull` and `components push` commands, outlining their usage, options, and examples for managing Storyblok components.
- Updated the main README to reference the new `components` module and its subcommands, enhancing clarity for developers.
- Adjusted option descriptions for consistency, including the removal of default values where applicable.

alvarosabu committed:
docs(components): remove trailing whitespace from README files

- Cleaned up the README files for the `components`, `components pull`, and `components push` commands by removing unnecessary trailing whitespace to improve formatting and maintain consistency across documentation.
- Ensured all files end with a newline for better compatibility with various text editors.

alvarosabu committed:
docs(migrations): add README files for migrations commands

- Created detailed README files for the `migrations`, `migrations generate`, and `migrations rollback` commands, outlining their usage, options, and examples for managing Storyblok component migrations.
- Updated the `migrations run` command documentation to include new options for publication modes and improved examples for clarity.
- Enhanced overall documentation structure to provide comprehensive guidance for developers interacting with the migrations functionality.

jeremenichelli commented on README.md:
```diff
@@ -48,7 +48,10 @@
 
-For complete documentation, please visit [https://storyblok.com/docs](https://storyblok.com/docs/packages/my-joyful-package)
+> [!WARNING]
+> Official documentation about this package v4 is not yet available. Please refer to the internal documentation for the [v4 beta](src/README.md).
```
```suggestion
> Official documentation for this package v4 is still in development. In the meantime, please refer to the internal documentation for the [v4 beta](src/README.md).
```

jeremenichelli commented on README.md:
```diff
@@ -48,7 +48,10 @@
+> [!WARNING]
+> Official documentation about this package v4 is not yet available. Please refer to the internal documentation for the [v4 beta](src/README.md).
+
+~~For complete documentation, please visit [https://storyblok.com/docs](https://storyblok.com/docs/packages/my-joyful-package)~~
```
I would comment this out until we have the actual documentation link.

jeremenichelli commented on src/README.md:
```diff
@@ -0,0 +1,53 @@
+![Storyblok ImagoType](https://raw.githubusercontent.com/storyblok/.github/refs/heads/main/profile/public/github-banner.png)
+
+# Storyblok CLI Package Reference
```
```suggestion
# Storyblok CLI
```

jeremenichelli commented on src/README.md:
````diff
@@ -0,0 +1,53 @@
+
+## Installation
+
+For the latest beta version, install the package using the following command:
+
+```bash
+npm install storyblok@beta
+```
+
+Or for an specific beta version:
+
+```bash
+npm install storyblok@4.0.0-beta.<version>
+```
````
My suggestion is to keep the non-beta installation command in here. It makes sense for people to want the stable version for a new project.

```suggestion
Add the package to your project with the following command:

```bash
npm install storyblok@latest
```

You can also install the beta version, currently in development:

```bash
npm install storyblok@beta
```

It's also possible to target a specific beta version:

```bash
npm install storyblok@4.0.0-beta.<version>
```
```

manuelschroederdev replied:
Hey @alvarosabu, is this readme going to be in `main`, or in a beta branch? If it's supposed to be in `main`, I agree with @jeremenichelli.

alvarosabu replied:
No, it's going to live on the `next` branch @manuelschroederdev @jeremenichelli these documentations are strictly for the beta since they will be replaced by official documentation later on

manuelschroederdev replied:
That's what I thought. 😄 Then feel free to disregard the suggestion from Jeremias here.

jeremenichelli commented on src/README.md:
```diff
@@ -0,0 +1,53 @@
+| `--ci` | Enable CI mode (coming soon) | `false` |
+
+> [!TIP]
+> When reporting a bug or opening a support ticket, please run the command with the `--verbose` flag to provide detailed information that will help us better understand and resolve the issue.
```
```suggestion
> When reporting a bug or opening a support ticket, please run the command with the `--verbose` flag and add the output to it. This will help us better understand and resolve the issue.
```

jeremenichelli commented on src/README.md:
```diff
@@ -0,0 +1,53 @@
+
+## Global Options
+
+These options can be used with any command:
```
```suggestion
These options are available for all commands:
```

jeremenichelli commented on src/README.md:
````diff
@@ -0,0 +1,53 @@
+npm install storyblok@4.0.0-beta.<version>
+```
+
+## Command Reference
````
Probably `## API` is a better title for this section.

jeremenichelli commented on src/README.md:
```diff
@@ -0,0 +1,53 @@
+| [`migrations run`](./commands/migrations/run/README.md) | ✅ Ready | Replaces previous run-migrations |
+| [`migrations rollback`](./commands/migrations/rollback/README.md) | ✅ Ready | Replaces previous rollback-migrations |
+| [`types generate`](./commands/types/generate/README.md) | ✅ Ready | Replaces previous generate-typescript-typedefs |
+| `sync` | ⚠️ v3 | |
```
I suggest leaving only the emojis, and adding a legend in the end or before the table.

manuelschroederdev replied:
@jeremenichelli I'm also not a fan of emojis, but I think they serve a purpose here.

alvarosabu replied:
My intention with the emojis was to help the user to quickly identify the status visually when reading through the table. Normally it would be done with CSS but we don't have that flavor on markdown.

Considering this is temporal documentation, I wouldn't be strict about this part.

jeremenichelli commented on src/commands/components/pull/README.md:
````diff
@@ -0,0 +1,191 @@
+        └── tags.json           # Component tags
+```
+
+3. Pull components with custom filename:
````
```suggestion
3. Pull components with a custom file name:
```

jeremenichelli replied:
If you want to keep the word join together then make it a code element.

jeremenichelli commented on src/commands/components/pull/README.md:
```diff
@@ -0,0 +1,191 @@
+
+Where:
+- `{path}` is the base path (default: `.storyblok`)
+- `{spaceId}` is your Storyblok space ID
```
```suggestion
- `{spaceId}` is your Storyblok Space ID
```

jeremenichelli commented on src/commands/components/push/README.md:
```diff
@@ -0,0 +1,164 @@
+The `components push` command allows you to upload components and their dependencies to your Storyblok space.
+
+> [!WARNING]
+> This command requires that you have previously used the `components pull` command to download the components. If you used any options during the pull (like `--suffix` or `--separate-files`), you must use the same options when pushing to ensure the files are found correctly.
```
```suggestion
> This command requires you have previously used the `components pull` command to download those components. If you used any flags during the pull (like `--suffix` or `--separate-files`), you must apply them with the same values when pushing to ensure files are found correctly.
```

jeremenichelli commented on src/commands/languages/README.md:
```diff
@@ -0,0 +1,100 @@
+# Languages Command
+
+The `languages` command allows you to manage languages in your Storyblok space. Currently, it supports pulling language configurations from your space.
```
```suggestion
The `languages` command allows you to handle internalization features within your Storyblok space. Currently, it supports pulling language configurations from your space.
```

manuelschroederdev replied:
Hey @jeremenichelli @alvarosabu, I think both versions are a bit too bold. As it doesn't allow for _managing_ anything at this point, we shouldn't raise any unrealistic expectations. If I'm not mistaken, this currently allows you to retrieve all configured languages as a JSON, similar to the `spaces/me` endpoint, correct?

alvarosabu replied:
Correct, it only allows pulling the languages available on the space @manuelschroederdev

manuelschroederdev replied:
Are there any plans to expand the functionality in the future?

alvarosabu replied:
@manuelschroederdev not in near future

jeremenichelli commented on src/commands/migrations/run/README.md:
```diff
@@ -1,96 +1,129 @@
 
-## General
+> [!NOTE]
+> Before running migrations, you need to create them first using the `migrations generate` command. See [migrations generate](./generate/README.md) for more details.
```
```suggestion
> Before running migrations, you must create them by running the `migrations generate` command. See [migrations generate](./generate/README.md) for more details.
```

jeremenichelli commented on src/commands/migrations/run/README.md:
```diff
@@ -1,96 +1,129 @@
-- [ ] It should show the command title
-- [ ] It should throw an error if the user is not logged in `You are currently not logged in. Please login first to get your user info.`
+> [!NOTE]
+> When running a migration, a snapshot of the story's content will be automatically created in the `.storyblok/migrations/{spaceId}/rollbacks` directory with a timestamp. These snapshots can be used to rollback changes if needed.
```
```suggestion
> When running a migration, a snapshot of the story's content will be automatically created in the `.storyblok/migrations/{spaceId}/rollbacks` directory with a timestamp. These can be used to rollback changes if needed.
```

jeremenichelli commented on src/commands/types/generate/README.md:
```diff
@@ -1,320 +1,82 @@
-- TypeScript installed in your project (if you plan to use the generated types)
-- Pull components from the target space using the `storyblok components pull` command
+> [!WARNING]
+> Before generating types, you must first pull your components using the `components pull` command. Make sure to use the same flags (`--separate-files`, `--suffix`) that you used when pulling components to ensure the types are generated correctly.
```
```suggestion
> Before generating types, first pull your components using the `components pull` command. Make sure to use the same flags (`--separate-files`, `--suffix`) that you used when pulling components to ensure the types are generated correctly.
```

jeremenichelli commented on src/commands/user/README.md:
```diff
@@ -0,0 +1,37 @@
+# User Command
+
+The `user` command allows you to view information about your currently logged-in Storyblok account.
```
```suggestion
This command allows you to view information about your currently logged-in Storyblok account.
```

manuelschroederdev commented on src/commands/logout/README.md:
```diff
@@ -0,0 +1,28 @@
+
+- The command will show a warning if you're already logged out
+- All stored credentials are removed from your system
+- You'll need to login again to use other Storyblok CLI commands
```
to log in (two words)

manuelschroederdev commented on src/commands/migrations/generate/README.md:
```diff
@@ -0,0 +1,107 @@
+
+## Notes
+
+- You must be logged in to use this command
```
Isn't that true for _every_ command? Maybe it doesn't have to be mentioned per command?

alvarosabu committed:
docs: apply suggestions from code review

Co-authored-by: Jeremias Menichelli <jmenichelli@gmail.com>

alvarosabu committed:
docs: update README files for improved clarity and structure

- Removed outdated documentation reference from the main README.
- Renamed the "Command Reference" section to "API" in the src README for better alignment with content.
- Enhanced the API section by adding a status legend to clarify the readiness of commands.
- Updated the description of the `languages` command to specify its functionality more accurately.

These changes aim to improve the overall documentation quality and user experience for developers interacting with the Storyblok CLI.

alvarosabu committed:
docs: remove unnecessary blank line in README

- Eliminated an extra blank line in the main README to improve formatting and maintain consistency across documentation. This minor adjustment enhances readability for developers.

alvarosabu committed:
docs: remove redundant login requirement note from multiple README files

- Eliminated the note stating "You must be logged in to use this command" from the README files of various commands, including `pull`, `push`, `languages`, `migrations`, and `user`. This change aims to streamline the documentation and reduce redundancy, as the login requirement is already implied in the context of command usage.

edodusi reviewed (approved):
@alvarosabu I don't have much to add here, to me the README are fine and I more or less have tested all the commands
